src/Map/Atlas.model.js

```diff
@@ -0,0 +1,232 @@
+import { Model } from 'decentraland-commons'
+
+import { ParcelLocation } from './ParcelLocation'
+import { ParcelReference } from './ParcelReference'
+import { Asset, Estate, Parcel, ParcelQueries } from '../Asset'
+import { Contribution } from '../Contribution'
+import { District } from '../District'
+import { Publication } from '../Publication'
+import { SQL, raw } from '../database'
+import { asyncPool } from '../lib'
+import { isDistrict } from '../../shared/district'
+import { isEstate } from '../../shared/parcel'
+import { ASSET_TYPES } from '../../shared/asset'
+import { PUBLICATION_STATUS } from '../shared/publication'
+
+export class Atlas extends Model {
+  static tableName = 'atlas'
+  static columnNames = [
+    'id',
+    'x',
+    'y',
+    'estate_id',
+    'district_id',
+    'owner',
+    'price',
+    'name',
+    'type',
+    'asset_type',
+    'is_connected_left',
+    'is_connected_top',
+    'is_connected_topleft'
+  ]
+
+  static async upsertAsset(assetId, assetType) {
+    const asset = await Asset.getModel(assetType).findOne(assetId)
+
+    switch (assetType) {
+      case ASSET_TYPES.parcel:
+        return this.upsertParcel(asset)
+      case ASSET_TYPES.estate:
+        return this.upsertEstate(asset)
+      default:
+        throw new Error(`The asset type ${assetType} is invalid`)
+    }
+  }
+
+  static async upsertEstate(estate) {
+    return asyncPool(
+      estate.data.parcels,
+      ({ x, y }) =>
+        Parcel.findOne({ x, y }).then(parcel => this.upsertParcel(parcel)),
+      10
+    )
+  }
+
+  static async upsertParcel(parcel) {
+    const now = new Date()
+    const row = await this.buildRow(parcel)
+
+    row.created_at = now
+    row.updated_at = now
+
+    const values = Object.values(row)
+
+    return this.db.query(
+      `INSERT INTO ${this.tableName}(
+       ${this.db.toColumnFields(row)}
+      ) VALUES(
+       ${this.db.toValuePlaceholders(row)}
+      ) ON CONFLICT (id) DO UPDATE SET
+        ${this.db.toAssignmentFields(row)};`,
+      values
+    )
+  }
+
+  static async inRange(topLeft, bottomRight) {
+    const columnNames = this.filterColumnNames([
+      'district_id',
+      'asset_type'
+    ]).join(', ')
+
+    return this.db.query(SQL`
+      SELECT ${raw(columnNames)}
+        FROM ${raw(this.tableName)}
+        WHERE ${ParcelQueries.whereIsBetweenCoordinates(topLeft, bottomRight)}`)
+  }
+
+  static async inRangePNG(topLeft, bottomRight) {
+    return this.db.query(SQL`
+      SELECT x, y, type, is_connected_left as left, is_connected_top as top, is_connected_topleft as "topLeft"
+        FROM ${raw(this.tableName)}
+        WHERE ${ParcelQueries.whereIsBetweenCoordinates(topLeft, bottomRight)}`)
+  }
+
+  static async inRangeFromAddressPerspective(topLeft, bottomRight, owner) {
```

Comment:
Maybe this method was there... Do you think it is a good opportunity to document what the params should be and what does the method?